src/idl_gen_rust.cpp

```diff
@@ -1501,6 +1501,17 @@ class RustGenerator : public BaseGenerator {
 
     // The root datatype accessors:
     code_ += "#[inline]";
+    code_ +=
+        "pub fn root_as_{{STRUCT_NAME_SNAKECASE}}<'a>(buf: &'a [u8])"
+        " -> {{STRUCT_NAME}}<'a> {";
+    code_ += "  flatbuffers::get_root::<{{STRUCT_NAME}}<'a>>(buf)";
+    code_ += "}";
+    code_ += "";
+
+    // legacy accessor
+
+    code_ += "#[inline]";
+    code_ += "#[deprecated(since = \"1.13.0\", note = \"getters starting with get_ are deprecated\")]";
```

Comment:
I think its important to add a date (and/or version) of deletion too, partly to remind us to do so, but also since things can be marked as deprecated yet stick around for a while so some people need a bit of a push.

Reply:
Sure, but the since field requires a semver version.

Reply:
Sorry, I was unclear. I mean you should write

```c
code_ += "#[deprecated(since = \"1.13.0\", note = \"getters starting with get_ are deprecated"
         " and will no longer be generated in the first flatc release after 2021-01-01.\")]";
```

because deletion is a separate event than deprecation that should be documented too (partly so we maintainers know when deleting the code is okay).

Reply:
@aardappel maybe removal from 1.14?

Reply:
yes that sounds fine to me

Reply:
This looks fixed.

Fwiw, I prefer dates to semver because we can't be too sure what the next semver string is or when it'll be released. I'm not too adamant on that position though.

Reply:
Dates have higher chances of change than versions, I think. Would be a bit weird if we lied in the deprecation message :)